### ISSUE TITLE
[Feature] User feedback: Postgres db performance enhancement + editor experience

### ISSUE THREAD

**Issue by sumitsum** [appsmithorg/appsmith#3387]
## Summary
User feedback around postgres db integration: 
- It is faster to use postgREST rather than db integration because direct API call is faster. 
- Query editor experience could be improved for very large queries. 

## Additional Context
Discord chat:
```
PostgREST allows me to add all the advantages of a proper API with very little overhead. It also lets me use appsmith as a prototype/internal use framework that shares APIs with the production application.

For the production application PostgREST has been noticably faster than using a bespoke API server. Also I have a lot of functions that execute a couple hundred lines of SQL which the editor is not really ready for.

```

**Comment by Nikhil-Nandagopal** [comment]
@sumitsum I think they are talking about it being faster than creating an API server from scratch. Can you verify this once?

**Comment by sumitsum** [comment]
> @sumitsum I think they are talking about it being faster than creating an API server from scratch. Can you verify this once?

@Nikhil-Nandagopal I have DMed the user for clarification.

**Comment by sumitsum** [comment]
> @sumitsum I think they are talking about it being faster than creating an API server from scratch. Can you verify this once?

@Nikhil-Nandagopal This is response I got from the user: 

```
I don't see a reason to use the SQL integrations at all since I already have a super-fast(~1ms overhead) api layer on top of my PostgreSQL instance. Java and C# have been noticeably slower for the same job
I've also been able to set up an A&A flow that integrates directly with PostgREST using keycloak so a public facing application would still be without access to data that should not be available.
Using a couple variables and 1 call I now have an inn app login screen in appsmith that implicitly transmits the logged in user to my PostgREST APIs
TLDR: the DB integration is not a section of appsmith that is relevant to my use case, I do see it's appeal for others.

I want to use the exact same set of APIs between my appsmith prototypes and my production grade application.
```

**Comment by Nikhil-Nandagopal** [comment]
Yeah sounds about right. So can we close this issue then?

**Comment by sumitsum** [comment]
@Nikhil-Nandagopal yes, I think we can close this since the query editor issue should be covered via https://github.com/appsmithorg/appsmith/issues/3368
